core/src/main/java/com/novoda/landingstrip/LandingStrip.java

```diff
@@ -153,9 +155,22 @@ private void notifyDataSetChanged(PagerAdapter pagerAdapter) {
         }
         ScrollingPageChangeListener scrollingPageChangeListener = new ScrollingPageChangeListener(state, tabsContainer,
                 scrollOffsetCalculator, this, fastForwarder, onPageChangeListenerCollection);
+        removeAllListenersFrom(viewPager);
+        trackAttachedListener(scrollingPageChangeListener);
         tabsContainer.startWatching(viewPager, scrollingPageChangeListener);
     }
 
+    private void removeAllListenersFrom(ViewPager viewPager) {
+        for (ScrollingPageChangeListener listener : onScrollingPageChangeListeners) {
+            viewPager.removeOnPageChangeListener(listener);
+        }
+        onScrollingPageChangeListeners.clear();
+    }
+
+    private void trackAttachedListener(ScrollingPageChangeListener scrollingPageChangeListener) {
+        onScrollingPageChangeListeners.add(scrollingPageChangeListener);
```

Comment:
because of the scope of this method you could simplify this var `scrollingPageChangeListener` to be just `listener`